static/app/components/searchSyntax/evaluator.tsx

```diff
@@ -99,30 +99,29 @@ function processTokenResults(tokens: TokenResult<Token>[]): ProcessedTokenResult
 }
 
 function isBooleanOR(token: ProcessedTokenResult): boolean {
-  return (
-    token && token.type === Token.LOGIC_BOOLEAN && token.value === BooleanOperator.OR
-  );
+  return token?.type === Token.LOGIC_BOOLEAN && token?.value === BooleanOperator.OR;
```

Comment:
Shouldn't line 101 be `ProcessedTokenResult | undefined`?

Reply:
Nah, this narrows down on the type. ProcessedTokenResult has different shapes, so this checks that it has a type and then narrows down to the boolean or as well.